google/cloud/forseti/notifier/notifier.py

```diff
@@ -57,16 +57,18 @@ def find_notifiers(notifier_name):
 # pylint: enable=inconsistent-return-statements
 
 
-def convert_to_timestamp(violations):
+def convert_to_timestamp(session, violations):
     """Convert violation created_at_datetime to timestamp string.
     Args:
+        session (object): session object to work on.
         violations (sqlalchemy_object): List of violations as sqlalchemy
             row/record object with created_at_datetime.
     Returns:
         list: List of violations as sqlalchemy row/record object with
             created_at_datetime converted to timestamp string.
     """
     for violation in violations:
+        session.expunge(violation)
```

Comment:
Thanks for the explanation.  It is just strange that the session needs to be touched here at all.  Perhaps we should make the list() return a non-sqlalchemy objects or turn-off auto-commit?

```Re. the expunge(): our sessions have autocommit enabled i.e. when they are destructed they try to flush all the "dirty" objects to the database and this results in "operational error" exceptions for violation objects whose created_at_datetime properties have been set to a string in the convert_to_timestamp() function.
Why? Because their column type is DateTime and not Text.```

Reply:
Turning off auto-commit would be quite a sweeping change. Make `list()` return non-sqlalchemy objects is probably better.

What's wrong with `expunge()` BTW?

Reply:
I mentioned that this method's purpose should be to do a simple conversion to timestamp string.  If the session needs to be managed here, it is an unrelated operation, and perhaps likely that sqlalchemy data type should not be used here.